Feature/data config file

roje-bodc opened the pull request:
initial configuration file docs.

davbyr reviewed (approved):
Look good to me. I have a couple of optional comments/questions:

1. It might good to specify which 'side' of the dictionaries end up in the COAsT object (key or item). It's fairly clear anyway, but just saying it explicitly would be useful (unless it's already there and I've missed it).
2. Can chunks be passed as a dictionary of dimensions:numbers as well? I thought I saw this in a branch at some point. There may be different dimension orders in some variables and then the order in the list might not be same applied across them. 

EDIT: The second point isn't really relevant to this pull request though sorry. I've left it there anyway.

roje-bodc commented:
> Look good to me. I have a couple of optional comments/questions:
> 
> 1. It might good to specify which 'side' of the dictionaries end up in the COAsT object (key or item). It's fairly clear anyway, but just saying it explicitly would be useful (unless it's already there and I've missed it).
> 2. Can chunks be passed as a dictionary of dimensions:numbers as well? I thought I saw this in a branch at some point. There may be different dimension orders in some variables and then the order in the list might not be same applied across them.
> 
> EDIT: The second point isn't really relevant to this pull request though sorry. I've left it there anyway.

1) Each key mentioned in the table is added to the config object, and the values are anything that isn't mentioned in the table. I was thinking that since the user shouldn't have to interact directly with the config object themselves anyhow, then they don't really need to understand exactly how the data is stored? Or is the assumption that they will be using the config object themselves?

2) Good spot, I have updated the docs now to mention that chunks should be passed in as a dict.  There were a few inconsistencies with the package code too which I have fixed and will push to the coast repo (just some stuff with doctstrings/type-hinting still mentioning chunks should be a list.)

jpolton commented on content/en/docs/Examples/configuration_files.md:
```diff
@@ -0,0 +1,181 @@
+| ``processing_flags`` | A list of strings referring to any preliminary processing methods to be carried out on the data. |
+
+### Example configuration file
+Below is the template of a gridded configuration file:
```
where "ex[number]" represents a static variable name that is hardwired into the package. E.g. "temperature".


(I find the use of non-standard variable names to demonstrate standard variable names a bit misleading. This caveat might clarify the situation)

jpolton commented on content/en/docs/Examples/configuration_files.md:
````diff
@@ -0,0 +1,181 @@
+	]
+}
+```
+Raw __example configuration__ files can be found in the ``config/`` directory within the [COAsT](https://github.com/British-Oceanographic-Data-Centre/COAsT/tree/master/config) github repository.
````
As I recall the files in config/ are also templates rather than 'working' or 'raw' files?

davbyr reviewed (approved):
@roje-bodc I think some users will interact with the config files, so it's probably good to make sure it is explicitly said somewhere. Jeff's suggestions works.

But otherwise looks great.
